almkanal/almkanal.py

```diff
@@ -0,0 +1,629 @@
+from pathlib import Path
+
+import attrs
+import mne
+import numpy as np
+import pandas as pd
+from attrs import define, field
+from numpy.typing import ArrayLike, NDArray
+
+from almkanal.data_utils.check_data import check_raw_epoch
+from almkanal.data_utils.data_classes import ICAInfoDict, InfoClass, PickDictClass
+from almkanal.preproc_utils.bio_utils import run_bio_preproc
+from almkanal.preproc_utils.ica_utils import run_ica
+
+# all them utility functions
+from almkanal.preproc_utils.maxwell_utils import run_maxwell
+from almkanal.src_utils.headmodel_utils import compute_headmodel, make_fwd
+from almkanal.src_utils.src_utils import comp_spatial_filters, src2parc
+
+
+@define
+class AlmKanal:
```

Comment:
As this is **the** one and only entry point for users, the documentation for this class needs to be expanded a lot. Especially as there are lots of attributes that must be set, can be set, should be set, must not be set, depending on the state of the preprocessing that might have been done outside.